### PR TITLE
Some changes for FreeBSD

### PR DESCRIPTION
**Pull request by RenZ0**
The first commit is for FreeBSD 10 (old one).
The second one is for FreeBSD 10.4 and 11.2, work in progress.

### CODE REVIEW

**Comment by peternewman** [comment]
Thanks, do you want to target the 0.10 branch please @RenZ0 ?

File: common/testing/Makefile.mk [hunk 1]
```diff
@@ -6,5 +6,6 @@ noinst_LTLIBRARIES += common/testing/libolatesting.la \
 common_testing_libolatesting_la_SOURCES = \
     common/testing/MockUDPSocket.cpp \
     common/testing/TestUtils.cpp
+common_testing_libolatesting_la_CXXFLAGS = $(COMMON_TESTING_PROTOBUF_FLAGS)
```
Comment (peternewman) [inline common/testing/Makefile.mk]: What was the exact message you were getting for this please and I'll see if we can use a more bespoke and appropriate set of flags?

Comment (RenZ0) [reply common/testing/Makefile.mk]: It's on a big machine and I will start it asap, say in a week or so.

**Comment by peternewman** [comment]
This change is also breaking the FreeBSD build: https://github.com/OpenLightingProject/ola/pull/1226

**Comment by peternewman** [comment]
Superseded by #1460 0

**Pull request closed without merging**